compiler/testData/codegen/box/when/exhaustiveBoolean.kt

```diff
@@ -1,4 +1,3 @@
-// IGNORE_BACKEND_FIR: JVM_IR
```

Comment:
This is, well, very side-case, and I'm not sure we should do all of this just to support it. Exhaustive when on boolean is rare by itself, and using expressions instead `true` and `false` there is even more rare. https://youtrack.jetbrains.com/issue/KT-39883 created, to be discussed.

Reply:
Another case when IR evaluator can't help us is FIR serializer, also to be discussed.